### PR TITLE
Flavor Update

### PR DESCRIPTION
**Pull request by PidgeyThePirate**
**What does this PR do:**
The system in SS13 which describes tastes and flavours is slightly underdeveloped. I have attempted to add a bit of flair to some of the various reagents, whether they be food, drink or something you probably shouldn't be eating. Grants a few more options to chefs and barkeeps.

(almost) every reagent that tastes of "bitterness" or some other generic string has been assigned a new flavor; flavor strength has been tentatively used with salt, msg and most toxins; salt and MSG are more likely to drown out other flavors in large amounts, while toxins are much harder to detect if masked correctly (they are roughly 8 times weaker than MSG, meaning that toxin and MSG in roughly equal proportions will mask the taste of poison, or salt:toxin in a 2:1 ratio). In addition, nutriment's flavor has been changed from "bitterness" to null, making basically everything in the kitchen a little less bitter.

[Thread](https://nanotrasen.se/forum/topic/14569-flavor-update/?tab=comments)
[Spreadsheet](https://docs.google.com/spreadsheets/d/1-Ou9mZtTqLxYGxmtRofNgZl7GRl7jIY6PDFZBAjT80w/)

**Changelog:**
:cl:
tweak: changed a whole bunch of strings to add a bit more flavor to most reagents in the game (most of these can be found in the spreadsheet).
experiment: resurrected the old "flavor_strength" value which appears fully functional in testing but has been entirely unused.
/:cl:

### CODE REVIEW

**Comment by variableundefined** [comment]
tg's taste system is much easier to expand on. Do you think you can port it? It is not super easy for a first time PR, however.

https://github.com/ParadiseSS13/Paradise/pull/10331

**Comment by alex-gh** [comment]
I don't think it is a good idea to add explicit taste messages to all the harmful reagents.

It is too easily meta-able and punishes stealthy antags even more.

**Comment by PidgeyThePirate** [comment]
I can certainly try. The scope is a bit beyond what I originally had in mind, but I'm relatively familiar with the taste system and I should be able to fumble my way through it... maybe.

**Comment by TheMadTrickster** [comment]
this actually interests me a lot lol, hope this gets given a fair chance

**Comment by KasparoVy** [comment]
The snozzberries taste like snozzberries.

**Comment by variableundefined** [comment]
I'll possibly tackle this myself in the next few days, probably the weekend, if you don't mind.

**Comment by PidgeyThePirate** [comment]
By all means have at it. You'd probably do it better than I would.

File: code/modules/reagents/chemistry/reagents/water.dm [hunk 1]
```diff
@@ -339,7 +340,7 @@
 	drink_icon = "glass_clear"
 	drink_name = "Glass of Water"
 	drink_desc = "The father of all refreshments."
-	taste_message = null
+	taste_message = "holiness"
```
Comment (Citinited) [inline code/modules/reagents/chemistry/reagents/water.dm]: You shouldn't be able to distinguish holy water from normal water by merely tasting it

Comment (PidgeyThePirate) [reply code/modules/reagents/chemistry/reagents/water.dm]: That can certainly be changed, though I don't really think it matters personally. If it matters to you, the fact that you're tasting it means it's already waaay too late. It should taste like hellfire to creatures it harms anyway.

**Comment by Fox-McCloud** [comment]
For reference, anything that's not explicitly a drink of consumable really shouldn't have a taste message.

Taste messages are mostly for fluff. We really don't want players sussing out if they've been poisoned, or not, based on how something tastes (if the chef happened to want to inject his food with poison).

**Comment by PidgeyThePirate** [comment]
That's what the taste_strength variable is there for. It's to make the poison easy to mask with other tastes. It's currently sitting at 0.5, we could probably make it more like 0.2 if that proves to be an issue.

**Comment by variableundefined** [comment]
I think it's better to make them completely tasteless so that poison isn't further nerfed - poisoning food is already a rare strategy and I don't think it needs to be punished further.

**Comment by variableundefined** [comment]
@PidgeyThePirate I approve this PR but what Fox said need to be addressed first - Keep taste message off nanomachines, poison, etc.

**Comment by PidgeyThePirate** [comment]
Alright, I'll revert those files to keep what we had before. Most of the poisons were tasteless in that.

**Comment by PidgeyThePirate** [comment]
That should do it... I think. Unless I screwed up the repo again.

**Pull request merged**